File: python/8kyu/abbrev_name.py
import codewars_test as test


def abbrev_name(name: str) -> str:
    return '.'.join(x[0].capitalize() for x in name.split(' '))


# def abbrevName(name):
#     return '.'.join(w[0] for w in name.split()).upper()


@test.describe("Fixed Tests")
def basic_tests():
    @test.it('Basic Test Cases')
    def basic_test_cases():
        test.assert_equals(abbrev_name("Sam Harris"), "S.H")
        test.assert_equals(abbrev_name("patrick feenan"), "P.F")
        test.assert_equals(abbrev_name("Evan C"), "E.C")
        test.assert_equals(abbrev_name("P Favuzzi"), "P.F")
        test.assert_equals(abbrev_name("David Mendieta"), "D.M")

File: typescript/8kyu/abbrev_name/abbrev_name.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.abbrevName = void 0;
function abbrevName(name) {
    return name.split(" ").map(word => word.charAt(0).toUpperCase()).join(".");
}
exports.abbrevName = abbrevName;
const chai_1 = require("chai");
describe("Sample Test Cases", function () {
    const SamHarris = abbrevName("Sam Harris");
    const PatrickFeenan = abbrevName("Patrick Feenan");
    const RFavuzzi = abbrevName("R Favuzzi");
    it("Should return a proper abbreviate", function () {
        chai_1.assert.strictEqual(SamHarris, "S.H", "Should return S.H");
        chai_1.assert.strictEqual(PatrickFeenan, "P.F", "Should return P.F");
        chai_1.assert.strictEqual(RFavuzzi, "R.F", "Should return R.F");
    });
});
